internal/appsec/appsec.go

```diff
@@ -9,42 +9,60 @@
 package appsec
 
 import (
+	"os"
 	"sync"
 
+	rc "github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+
 	"gopkg.in/DataDog/dd-trace-go.v1/internal/appsec/dyngo"
 	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/remoteconfig"
 )
 
 // Enabled returns true when AppSec is up and running. Meaning that the appsec build tag is enabled, the env var
 // DD_APPSEC_ENABLED is set to true, and the tracer is started.
 func Enabled() bool {
 	mu.RLock()
 	defer mu.RUnlock()
-	return activeAppSec != nil
+	return activeAppSec != nil && activeAppSec.started
 }
 
 // Start AppSec when enabled is enabled by both using the appsec build tag and
 // setting the environment variable DD_APPSEC_ENABLED to true.
-func Start() {
+func Start(opts ...StartOption) {
 	enabled, err := isEnabled()
 	if err != nil {
 		logUnexpectedStartError(err)
 		return
 	}
-	if !enabled {
-		log.Debug("appsec: disabled by the configuration: set the environment variable DD_APPSEC_ENABLED to true to enable it")
-		return
-	}
 
 	cfg, err := newConfig()
 	if err != nil {
 		logUnexpectedStartError(err)
 		return
 	}
+	for _, opt := range opts {
+		opt(cfg)
+	}
 	appsec := newAppSec(cfg)
-	if err := appsec.start(); err != nil {
-		logUnexpectedStartError(err)
-		return
+
+	if !enabled {
+		// Check if the env var is set. If it is, appsec is specifically disabled. If not, it is disabled but can be
+		// enabled through remote config, and we should start the rc client
+		if _, set := os.LookupEnv(enabledEnvVar); set {
+			log.Debug("appsec: disabled by the configuration: set the environment variable DD_APPSEC_ENABLED to true to enable it")
+			return
+		}
+		log.Debug("appsec: %s is not set. AppSec won't start until activated through remote configuration", enabledEnvVar)
+	} else { // AppSec is specifically enabled
+		if err := appsec.start(); err != nil {
+			logUnexpectedStartError(err)
+			return
+		}
+	}
+	if appsec.rc != nil {
+		// TODO: register ASM_FEATURES callback
```

Comment:
Does this TODO need to be done before merging this?

Reply:
This will be addressed in a follow-up PR, the draft of which can be found [here](https://github.com/DataDog/dd-trace-go/pull/1489).